### PR TITLE
cleanup(tgc): Remove tfplan.json files

### PR DESCRIPTION
**Pull request by iyabchen**
Remove tfplan.json files for tgc unit tests. The test will generate the .tfplan.json files from the .tf files. 
Hence, the "unit" tests will require terraform binary execution.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by melinath — APPROVED** [review]
I've confirmed that the same tests are running beforehand and afterward.

It turns out that there's significant overhead from generating the terraform plans (even without network access) - on my machine it goes from about 1.6 seconds using the tfplan.json files to about 215 seconds generating them at test time. Looks like the runtime is similar in GHA.

I think that's acceptable for now - still only a few minutes. If it becomes an issue in the future we could likely generate the tfplan.json at generation time instead of test runtime - but that would potentially introduce other challenges.

**Comment by melinath** [comment]
I'd like to request a second reviewer on this since I'm pretty close to these changes.

**Comment by melinath** [comment]
@NickElliot context for this change go/tgc-tf-dev-override

File: mmv1/third_party/tgc/tests/source/read_test.go.erb [hunk 1]
```diff
@@ -50,7 +50,27 @@ func TestReadPlannedAssetsCoverage(t *testing.T) {
...
+				err = os.Rename(
+					filepath.Join(dir, c.name+".tfstate"),
+					filepath.Join(dir, "terraform.tfstate"),
+				)
+				if err != nil {
+					t.Fatalf("renaming tfstate: %v", err)
+				}
+			}
+
+      // Run terraform init and terraform apply to generate tfplan.json files
```
[the remark this answers is not in the document]
Comment (iyabchen) [reply mmv1/third_party/tgc/tests/source/read_test.go.erb]: Done.

File: mmv1/third_party/tgc/tests/source/read_test.go.erb [hunk 2]
```diff
@@ -50,7 +50,27 @@ func TestReadPlannedAssetsCoverage(t *testing.T) {
 			defer os.RemoveAll(dir)
 
 			generateTestFiles(t, "../testdata/templates", dir, c.name+".json")
-			generateTestFiles(t, "../testdata/templates", dir, c.name+".tfplan.json")
+			generateTestFiles(t, "../testdata/templates", dir, c.name+".tf")
+
+			// tfstat files are for cases testing updates, eg. project update.
```
[the remark this answers is not in the document]
Comment (iyabchen) [reply mmv1/third_party/tgc/tests/source/read_test.go.erb]: Done

File: mmv1/third_party/tgc/tests/source/read_test.go.erb [hunk 3]
```diff
@@ -114,7 +134,10 @@ func TestReadPlannedAssetsCoverage_WithoutDefaultProject(t *testing.T) {
 			defer os.RemoveAll(dir)
 
 			generateTestFiles(t, "../testdata/templates", dir, c.name+"_without_default_project.json")
-			generateTestFiles(t, "../testdata/templates", dir, c.name+".tfplan.json")
+			generateTestFiles(t, "../testdata/templates", dir, c.name+".tf")
+
+      // Run terraform init and terraform plan to generate tfplan.json files
```
[the remark this answers is not in the document]
Comment (iyabchen) [reply mmv1/third_party/tgc/tests/source/read_test.go.erb]: Done

**Review by melinath — APPROVED** [review]
LGTM pending review from @NickElliot

**Comment by iyabchen** [comment]
> .tf.json

I don't quite follow. All .tfplan.json files are removed from the testdata folder in this CL.

**Comment by iyabchen** [comment]
> Does this remaining reference to .tf.json need to be removed as well?
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/3b735dddf2c5987b98b4c06084bee51d8cbdd203/mmv1/provider/terraform_tgc.rb#L88

I don't quite follow. All .tfplan.json files are removed from the testdata folder in this CL.

**Comment by NickElliot** [comment]
Exactly -- so I wanted to verify if the linked validator should no longer be checking for .tf.json test fo;es (as they no longer are a part of the code). If the code is dead, it should be removed.

**Comment by NickElliot** [comment]
Ah, nevermind I see the problem, I thought I had been looking at the branch when I saw that but I accidentally started searching on main and the linked line was removed correctly in the first place. LGTM!

**Review by NickElliot — APPROVED** [review]

**Pull request merged**